params/variants.toml

```diff
@@ -5355,25 +5355,66 @@ sampler = "ф"
 samplerExplain = "Cyrillic Lower Ef"
 tagKind = "letter"
 
-[prime.cyrl-ef.variants.serifless]
+[prime.cyrl-ef.variants-buildup]
+entry = "bowl"
+descriptionLeader = "Cyrillic Lower Ef (`ф`)"
+
+[prime.cyrl-ef.variants-buildup.stages.bowl."*"]
+next = "bar"
+
+[prime.cyrl-ef.variants-buildup.stages.bowl.standard]
 rank = 1
-description = "Cyrillic Lower Ef (`ф`) with standard shape and without serifs"
-selector."cyrl/ef"  = "serifless"
+keyAffix = ""
+selectorAffix."cyrl/ef" = ""
+selectorAffix."cyrl/ef.BGR" = ""
 
-[prime.cyrl-ef.variants.cursive]
+[prime.cyrl-ef.variants-buildup.stages.bowl.split]
 rank = 2
-description = "Cyrillic Lower Ef (`ф`) with cursive shape"
-selector."cyrl/ef"  = "cursive"
+nonBreakingVariantAdditionPriority = 100
+descriptionAffix = "split bowl"
+selectorAffix."cyrl/ef" = "splitBowl"
+selectorAffix."cyrl/ef.BGR" = ""
 
-[prime.cyrl-ef.variants.top-serifed]
-rank = 3
-description = "Cyrillic Lower Ef (`ф`) with standard shape and serif at top"
-selector."cyrl/ef"  = "topSerifed"
+[prime.cyrl-ef.variants-buildup.stages.bar."*"]
+next = "serifs"
 
-[prime.cyrl-ef.variants.serifed]
-rank = 4
-description = "Cyrillic Lower Ef (`ф`) with standard shape and serifs at top and bottom"
-selector."cyrl/ef"  = "serifed"
+[prime.cyrl-ef.variants-buildup.stages.bar.straight]
+rank = 1
+keyAffix = ""
+descriptionAffix = "straight bar"
+selectorAffix."cyrl/ef" = ""
+selectorAffix."cyrl/ef.BGR" = ""
+
+[prime.cyrl-ef.variants-buildup.stages.bar.cursive]
+rank = 2
+next = "END"
+descriptionAffix = "cursive bar"
+selectorAffix."cyrl/ef" = "cursive"
+selectorAffix."cyrl/ef.BGR" = "cursive"
+
+[prime.cyrl-ef.variants-buildup.stages.serifs.serifless]
+rank = 1
+keyAffix = ""
```

Comment:
If `[prime.cyrl-ef.variants-buildup.stages.bar.cursive]` do terminate (i.e. not extending into the serifs level) then we could remove this line to make sure that variant names are kept unchanged.